cmd/build.go

```diff
@@ -13,7 +13,7 @@ import (
 
 func init() {
 	root.AddCommand(buildCmd)
-	buildCmd.Flags().StringP("builder", "b", "default", "Buildpack builder, either an as a an image name or a mapping name as defined in func.yaml")
+	buildCmd.Flags().StringP("builder", "b", "", "Buildpack builder, either an as a an image name or a mapping name as defined in func.yaml\nIf the value is empty then the value from the config will be used.")
```

Comment:
This is tricky to explain concisely.  In particular, the idea that func.yaml is more of a side-effect, but can be used to override, leads me to believe we should perhaps first highlight the purpose of writing to func.yaml in the first place:  to allow subsequent builds to remember earlier decisions:

```suggestion
	buildCmd.Flags().StringP("builder", "b", "", "Buildpack builder, either an as a an image name or a mapping name.  /n Specified value is stored in func.yaml for subsequent builds.")
```
Or is this even more confusing?